src/borg/testsuite/__init__.py

```diff
@@ -54,6 +55,86 @@
     st_mtime_ns_round = -4  # only >1 microsecond resolution here?
 
 
+_symlinks_supported = None
+_fifos_supported = None
+_hardlinks_supported = None
+_utime_supported = None
+
+
+def generate_tempfile_path():
+    return os.path.join(tempfile.gettempdir(), 'borg-test-' + uuid.uuid4().hex)
```

Comment:
Q: Does the same caveat that NamedTemporaryFile has apply to mkstemp?

If so, it may be better to just use TemporaryDirectory and create a file with a static name in there instead. As a context manager it also handles cleanup with no extra code.